price/src/price_tracker.rs

```diff
@@ -1,31 +1,91 @@
 use anyhow::{anyhow, Error, Result};
-use chrono::{DateTime, Duration, Utc};
-use file_store::{FileInfo, FileStore, FileType, Settings};
+use chrono::{DateTime, Duration, TimeZone, Utc};
+use config::{Config, File};
+use file_store::{FileInfo, FileStore, FileType};
 use futures::stream::{StreamExt, TryStreamExt};
 use helium_proto::{BlockchainTokenTypeV1, Message, PriceReportV1};
-use std::collections::HashMap;
-use tokio::sync::watch;
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, path::Path};
+use tokio::sync::{mpsc, watch};
 
-type Prices = HashMap<BlockchainTokenTypeV1, u64>;
+struct Price {
+    price: u64,
+    timestamp: DateTime<Utc>,
+}
+
+impl TryFrom<&PriceReportV1> for Price {
+    type Error = anyhow::Error;
+
+    fn try_from(value: &PriceReportV1) -> Result<Self, Self::Error> {
+        Ok(Self {
+            price: value.price,
+            timestamp: Utc
+                .timestamp_opt(value.timestamp as i64, 0)
+                .single()
+                .ok_or_else(|| anyhow!("Invalid timestamp: {}", value.timestamp))?,
+        })
+    }
+}
+
+type Prices = HashMap<BlockchainTokenTypeV1, Price>;
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct Settings {
+    price_duration_minutes: u64,
+    file_store: file_store::Settings,
+}
+
+impl Settings {
+    pub fn new<P: AsRef<Path>>(path: P) -> Result<Self> {
```

Comment:
price tracker is still treated as a process running in the context of another application, right? how are we instantiating the settings for a price tracker in the context of the Verifier (for example) without passing a separate config file path to this function?

Reply:
that function may not be needed, I copied from another crate.

Reply:
Removed 64f7e2a